classes/user/DAO.php

```diff
@@ -356,6 +357,20 @@ public function deleteUnvalidatedExpiredUsers(Carbon $dateTillValid, array $excl
         return $users->count();
     }
 
+    /**
+     * Get publication ids that have a matching setting
+     */
+    public function getIdsBySetting(string $settingName, $settingValue): Enumerable
```

Comment:
I think you can use the existing `Collector` class in calling code rather than adding this new function. Off the top of my head, something like:
```php
$ids = Repo::user()->getCollector()
  ->filterBySettings(['setting name' => 'setting value'])
  ->getIds();
```